instagram/src/components/PostContainer/PostsPage.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+import SearchBar from '../SearchBar/SearchBar';
+
+class PostsPage extends Component {
+    constructor() {
+      super();
+      this.state = {
+        posts: [],
+        filteredPosts: []
+      };
+    }
+    componentDidMount () {
+      this.setState({posts: dummyData});
+    }
+    searchPostsHandler = e => {
+      const posts = this.state.posts.filter(p => {
+        if (p.username.includes(e.target.value)) {
+          return p;
+        } else {
+          return null;
+        }
+      }); 
+      this.setState({ filteredPosts: posts });
+    };
+  
+    render() {
+      return (
+        <div className="App">
+         <SearchBar
+            searchTerm={this.state.searchTerm}
+            searchPosts={this.searchPostsHandler}
+          />
+          <PostContainer
+            posts={
+              this.state.filteredPosts.length > 0
+                ? this.state.filteredPosts
+                : this.state.posts
```

Comment:
oh this is interesting...so if there are no posts that match your request, return all the posts - very cool.